components/properties/src/maps.rs

```diff
@@ -173,6 +173,44 @@ impl<'a, T: TrieValue> CodePointMapDataBorrowed<'a, T> {
         let set = self.map.get_set_for_value(value);
         CodePointSetData::from_code_point_inversion_list(set)
     }
+
+    /// Yields an [`Iterator`] returning ranges of consecutive code points that
+    /// share the same value in the [`CodePointMapData`].
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use core::ops::RangeInclusive;
+    /// use icu::properties::maps::CodePointMapData;
+    /// use icu_codepointtrie::CodePointMapRange;
+    /// use icu_codepointtrie::planes;
+    ///
+    /// let planes_trie = planes::get_planes_trie();
+    /// let cp_map_data = CodePointMapData::from_code_point_trie(planes_trie);
+    /// let cp_map = cp_map_data.as_borrowed();
+    ///
+    /// let mut ranges = cp_map.iter_ranges();
+    ///
+    /// for plane in 0..=16 {
+    ///     let exp_start = plane * 0x1_0000;
+    ///     let exp_end = exp_start + 0xffff;
+    ///     assert_eq!(
+    ///         ranges.next(),
+    ///         Some(CodePointMapRange {
+    ///             range: RangeInclusive::new(exp_start, exp_end),
+    ///             value: plane as u8
+    ///         })
+    ///     );
+    /// }
+    ///
+    /// // Hitting the end of the iterator returns `None`, as will subsequent
+    /// // calls to .next().
+    /// assert_eq!(ranges.next(), None);
+    /// assert_eq!(ranges.next(), None);
+    /// ```
+    pub fn iter_ranges(&self) -> CodePointMapRangeIterator<'_, T> {
```

Comment:
Done.  I did this for `iter_ranges()` for the CodePointMapData API here and the data struct `PropertyCodePointMapV1`.  Nothing analogous seems needed for `iter_ranges()` for set data because it already returns `impl Iterator<Item = RangeInclusive<u32>>`.